daemon/daemon.go

```diff
@@ -141,13 +141,12 @@ func (daemon *Daemon) Get(prefixOrName string) (*Container, error) {
 		return containerByName, nil
 	}
 
-	containerID, indexError := daemon.idIndex.Get(prefixOrName)
-	if indexError != nil {
-		// When truncindex defines an error type, use that instead
-		if strings.Contains(indexError.Error(), "no such id") {
+	containerID, err := daemon.idIndex.Get(prefixOrName)
+	if err != nil {
+		if truncindex.IsNoSuchID(err) {
```

Comment:
Is there a reason you created a func for this test instead of just doing a typecast check?